### PR TITLE
Finish CRUD support for Projects, Experiments, and Variations

### LINKED ISSUE #2: Implement CRUD for Projects/Experiments/Variations
I'm writing a CLI utility similar to your optimizely-cli project and am looking to implement push AND pull functionality for experiments, variations and projects, but there's no Update defined for Projects, only Create and Read. Can you finish implementing CRUD methods for at least Projects, Experiments, and Variations?

Thanks!

### PR DESCRIPTION
**Pull request by nselvidge**
Resolves #2

### CODE REVIEW

File: lib/OptimizelyClient.js [hunk 1]
```diff
@@ -150,8 +150,61 @@ OptimizelyClient.prototype.getProject = function(options) {
         'Token': this.apiToken,
         'Content-Type': 'application/json'
       }
-    })
+    });
   }
+  /**
+   * @public
+   * @name  OptimizelyClient#updateProject
+   * @since  0.1.0
```
Comment (nsheaps) [inline lib/OptimizelyClient.js]: `@since 0.2.0`

File: lib/OptimizelyClient.js [hunk 2]
```diff
@@ -323,6 +376,33 @@ OptimizelyClient.prototype.getExperimentByDescription = function(options) {
   })
 }
 
+  /**
+   *@pubilc
```
Comment (nsheaps) [inline lib/OptimizelyClient.js]: comment spacing doesn't match the above

File: lib/OptimizelyClient.js [hunk 3]
```diff
@@ -323,6 +376,33 @@ OptimizelyClient.prototype.getExperimentByDescription = function(options) {
...
+   *@name OptimizelyClient#deleteExperiment
+   *@since 0.0.1
```
Comment (nsheaps) [inline lib/OptimizelyClient.js]: `@since 0.2.0`

File: lib/OptimizelyClient.js [hunk 4]
```diff
@@ -429,3 +509,30 @@ OptimizelyClient.prototype.pushVariation = function(options) {
     this.createVariation(options);
 }
 module.exports = OptimizelyClient;
```
Comment (nsheaps) [inline lib/OptimizelyClient.js]: This is called twice and not at the end. Not a problem per se, just curious as to why.

**Comment by nsheaps** [comment]
Looks awesome. Thanks so much for finishing up the CRUD implementation!

Might I suggest a version bump and an `npm publish`?

**Comment by nselvidge** [comment]
I've amended those issues.

**Comment by nsheaps** [comment]
ship it

**Pull request merged**